backend/inspirehep/disambiguation/utils.py

```diff
@@ -109,3 +110,30 @@ def update_author_names(author, signatures):
         author["name"]["name_variants"] = name_variants
     author.update(dict(author))
     return author
+
+
+def normalize_lit_author_name(lit_author_name, author_name):
+    lit_author_name_tokens = re.split(", | ", lit_author_name)
+    lit_author_last_names = lit_author_name.split(",")[0].split(" ")
+    lit_author_first_names_reordered = []
+    lit_author_last_names_reordered = []
+
+    author_last_names = set(author_name.split(",")[0].split(" "))
+    author_first_names = (
+        set(author_name.split(",")[1].split(" ")) if "," in author_name else []
+    )
+
+    for name_idx, name in enumerate(lit_author_name_tokens):
+        if name in author_last_names:
+            lit_author_last_names_reordered.append(name)
+            continue
+        if name in author_first_names:
+            lit_author_first_names_reordered.append(name)
+            continue
+        if name_idx + 1 > len(lit_author_last_names):
+            lit_author_first_names_reordered.append(name)
+        else:
+            author_first_names.append(name)
```

Comment:
why is this (plus normalization) not sufficient?
```suggestion
        else:
            lit_author_first_names_reordered.append(name)
```

Reply:
so we assume that lit author name always matches one of the names from author?

Reply:
hmmm, good point. I was thinking we apply this only when there's only a single last name, but that doesn't necessarily match either. If there's no match among the last names, we should keep the current last name then (not sure that's what's happening here though).